clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2117,14 +2117,23 @@ private ClusterResourceListeners configureClusterResourceListeners(Deserializer<
     private void close(long timeoutMs, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
         AtomicReference<Throwable> firstException = new AtomicReference<>();
+
+        long closeStart = time.milliseconds();
         try {
             if (coordinator != null)
                 coordinator.close(time.timer(Math.min(timeoutMs, requestTimeoutMs)));
         } catch (Throwable t) {
             firstException.compareAndSet(null, t);
             log.error("Failed to close coordinator", t);
         }
-        ClientUtils.closeQuietly(fetcher, "fetcher", firstException);
+        long closeTimeoutLeft = timeoutMs - (time.milliseconds() - closeStart);
```

Comment:
wonder if we should handle cases where this is less than 0

Reply:
Can we not use the Timer that we pass to the Coordinator?

Reply:
We should consider using the Timer abstraction in the Fetcher methods too.

Reply:
Yes, agreed

Reply:
Although if we use the same `Timer` we ought to make the coordinator close with a timer of `timeoutMs`, not the min of `timeoutMs` and `requestTimeoutMs`, right?